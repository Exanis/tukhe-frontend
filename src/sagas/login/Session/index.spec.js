import watcherSaga, { sessionLoginSaga } from "./index";

it('Should have a watcher saga', () => {
    const gen = watcherSaga();

    expect(gen.next()).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
});

it('Should log user in when there is a token', () => {
    jest.spyOn(Storage.prototype, 'getItem');
    Storage.prototype.getItem.mockReturnValue('test');
    const gen = sessionLoginSaga();

    expect(gen.next()).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
    expect(Storage.prototype.getItem).toHaveBeenCalledWith('tukhe-token');
});

it('Should log user in when there is no token', () => {
    jest.spyOn(Storage.prototype, 'getItem');
    Storage.prototype.getItem.mockReturnValue(false);
    const gen = sessionLoginSaga();

    expect(gen.next()).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
    expect(Storage.prototype.getItem).toHaveBeenCalledWith('tukhe-token');
});

it('Should log user in when there is a token but it is null', () => {
    jest.spyOn(Storage.prototype, 'getItem');
    Storage.prototype.getItem.mockReturnValue('null');
    const gen = sessionLoginSaga();

    expect(gen.next()).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
    expect(Storage.prototype.getItem).toHaveBeenCalledWith('tukhe-token');
});
