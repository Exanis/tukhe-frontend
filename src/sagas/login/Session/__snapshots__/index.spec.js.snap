// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should have a watcher saga 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [
        [Function],
        [Function],
      ],
      "context": null,
      "fn": [Function],
    },
    "type": "FORK",
  },
}
`;

exports[`Should log user in when there is a token 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "action": Object {
        "payload": "test",
        "type": "user/setToken",
      },
      "channel": undefined,
    },
    "type": "PUT",
  },
}
`;

exports[`Should log user in when there is a token but it is null 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "action": Object {
        "payload": null,
        "type": "user/setToken",
      },
      "channel": undefined,
    },
    "type": "PUT",
  },
}
`;

exports[`Should log user in when there is no token 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "action": Object {
        "payload": null,
        "type": "user/setToken",
      },
      "channel": undefined,
    },
    "type": "PUT",
  },
}
`;
